src/baldr/graphtile.cc

```diff
@@ -325,12 +325,13 @@ GraphId GraphTile::GetTileId(const std::string& fname) {
   //run backwards while you find an allowed char but stop if not 3 digits between slashes
   std::vector<int> digits;
   auto last = pos;
-  for(--pos; pos >= 0; --pos) {
+  for(--pos; pos < last; --pos) {
+    auto& c = fname.at(pos);
```

Comment:
before i was using `operator[]` which doesnt do bounds checking, using `at` makes sure to throw if the thing goes out of range so at least we wont read out of bounds.